### DISCUSSION TITLE
Is there any way to pass contract address from `write` function in `useContractWrite`

### DISCUSSION THREAD

**Discussion by bigint** [wevm/wagmi#265] [Q&A]
Is there any way to pass contract address from `write` function in `useContractWrite`

something like

`write({ args: inputArray }, { addressOrName: '0x' })`

cc @tmm

**Comment by zkSoju** [comment]
I was just looking into this as well! Trying to call `setApprovalForAll` for multiple dynamic contracts is quite bothersome with a singular hook.

**Comment by bigint** [comment]
Any ideas @tmm 🙇🏻

**Reply by tmm** [reply]
Can consider, but we want hooks to be as declarative as possible. (This is especially true for an upcoming contract code generation feature and how `useContractWrite` works with that 🤐)
